lib/preprocessor.js

```diff
@@ -24,7 +24,13 @@ function executeProcessor (process, file, content) {
       }
     }
   })
-  return process(content, file, done) || donePromise
+
+  return (process(content, file, done) || Promise.resolve()).then((content) => {
+    if (content) {
```

Comment:
I found this code confusing to read.  Let's add a comment here:
`// async process correctly returned content/`